src/Gaffer/ContextVariablesTweaks.cpp

```diff
@@ -0,0 +1,122 @@
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/ContextVariablesTweaks.h"
+
+#include "IECore/DataAlgo.h"
+
+using namespace IECore;
+using namespace Gaffer;
+
+GAFFER_NODE_DEFINE_TYPE( ContextVariablesTweaks );
+
+size_t ContextVariablesTweaks::g_firstPlugIndex = 0;
+
+ContextVariablesTweaks::ContextVariablesTweaks( const std::string &name ) : ContextProcessor( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new BoolPlug( "ignoreMissing", Plug::In, false ) );
+	addChild( new TweaksPlug ( "tweaks" ) );
+}
+
+ContextVariablesTweaks::~ContextVariablesTweaks()
+{
+
+}
+
+Gaffer::BoolPlug *ContextVariablesTweaks::ignoreMissingPlug()
+{
+	return getChild<BoolPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::BoolPlug *ContextVariablesTweaks::ignoreMissingPlug() const
+{
+	return getChild<BoolPlug>( g_firstPlugIndex );
+}
+
+Gaffer::TweaksPlug *ContextVariablesTweaks::tweaksPlug()
+{
+	return getChild<TweaksPlug>( g_firstPlugIndex + 1 );
+}
+
+const Gaffer::TweaksPlug *ContextVariablesTweaks::tweaksPlug() const
+{
+	return getChild<TweaksPlug>( g_firstPlugIndex + 1 );
+}
+
+bool ContextVariablesTweaks::affectsContext( const Plug *input ) const
+{
+	return tweaksPlug()->isAncestorOf( input ) || input == ignoreMissingPlug();
+}
+
+void ContextVariablesTweaks::processContext( Context::EditableScope &context, IECore::ConstRefCountedPtr &storage ) const
+{
+	const TweaksPlug *tweaksPlug = this->tweaksPlug();
+	if( tweaksPlug->children().empty() )
+	{
+		return;
+	}
+
+	const bool ignoreMissing = ignoreMissingPlug()->getValue();
+
+	IECore::CompoundDataPtr sourceData = new CompoundData();
+	IECore::CompoundDataPtr result = new CompoundData();
```

Comment:
It took me a little while to realise that the only purpose of these variables was to keep their contents alive long enough. I've elaborated on that a bit below.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5047/commits/19418b4fa6215cc4afabe83fbd08f1c8acb24563